objects/src/assets/nonfungible.rs

```diff
@@ -204,11 +189,26 @@ impl Serializable for NonFungibleAsset {
 impl Deserializable for NonFungibleAsset {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
         let value: Word = source.read()?;
-
         Self::try_from(value).map_err(|err| DeserializationError::InvalidValue(err.to_string()))
```

Comment:
Shouldn't this also be updated? Basically, the round-trip of serializing/deserializing non-fungible assets should work even at this level.

Let's also add a test for this (unless one already exists).